.ci/static-checks.sh

```diff
@@ -537,8 +537,25 @@ check_docs()
 		fi
 
 		# Check the URL, saving it if invalid
-		( curl -sLf -o /dev/null "$url" ||\
-				echo "$url" >> "$invalid_urls") &
+		(
+			local http_statuses
+
+			# A single doc can generate multiple HTTP status
+			# codes, so save them all.
+			http_statuses=$(curl -sIL "$url" | grep -E "^HTTP" | awk '{print $2}' || true)
+
+			local status
+
+			for status in $http_statuses
+			do
+				local ret
+
+				# See https://en.wikipedia.org/wiki/List_of_HTTP_status_codes
+				{ echo "$status" | grep -qE "^(2|3)[0-9][0-9]"; ret=$?; } || true
+
+				[ $ret -ne 0 ] && { echo "$url" >> "$invalid_urls" && break; } || true
```

Comment:
You can somehow make this more readable with:
```bash
if ! echo "$status" | grep -qE "^(2|3)[0-9][0-9]"; then
	echo "$url" >> "$invalid_urls"
	break
fi
```

Reply:
Agreed - done.